gax-grpc/src/main/java/com/google/api/gax/grpc/UnaryCallable.java

```diff
@@ -328,4 +397,12 @@ public ResponseT call(RequestT request) {
     return new UnaryCallable<>(
         new BatchingCallable<>(callable, batchingDescriptor, batcherFactory), channel, settings);
   }
+
+  UnaryCallable<RequestT, ResponseT> withAuth(Credentials credentials) {
+    if (credentials == null) {
+      return this;
+    }
+    return new UnaryCallable<>(
+        new AuthCallable<>(callable, MoreCallCredentials.from(credentials)), channel, settings);
```

Comment:
I'd prefer for the `Credentials` -> `CallCredentials` conversion to be done inside `AuthCallable`, in order to confine the `CallCredentials` type as much as possible.